### PR TITLE
Upgrade toolchain to nightly-2024-03-15

### PR DESCRIPTION
**Pull request by celinval**
- Kani can no longer automatically detect loop bounds when using range.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: docs/src/getting-started/verification-results/src/main.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #[kani::proof]
+#[kani::unwind(4)]
```
Comment (feliperodri) [inline docs/src/getting-started/verification-results/src/main.rs:5]: Why we didn't need to unwind these cases before?

**Review by feliperodri — APPROVED** [review]

**Comment by adpaco-aws** [comment]
@karkhaz the test cases in this PR have regressed and Kani can no longer automatically unwind them.

File: docs/src/getting-started/verification-results/src/main.rs (hunk 1 above)
Comment (celinval) [reply docs/src/getting-started/verification-results/src/main.rs:5]: I still haven't had time to investigate why

**Comment by celinval** [comment]
I created https://github.com/model-checking/kani/issues/3088 to track the unwind issue.

File: docs/src/getting-started/verification-results/src/main.rs (hunk 1 above)
Comment (celinval) [reply docs/src/getting-started/verification-results/src/main.rs:5]: I created #3088 for us to investigate what can be done to remove these attributes.

**Pull request merged**